modules/snooker/src/main/ts/ephox/snooker/model/Fitment.ts

```diff
@@ -58,29 +59,42 @@ const measureHeight = (gridA: Structs.RowCells[], gridB: Structs.RowCells[]): De
   };
 };
 
-const generateElements = <T> (cells: T[], row: Structs.RowCells, generators: SimpleGenerators) => {
-  const getGenerator = row.section === 'colgroup' ? generators.col : generators.cell;
-  return Arr.map(cells, () => Structs.elementnew(getGenerator(), true));
+const generateElements = (amount: number, row: Structs.RowCells, generators: SimpleGenerators, isLocked: (idx: number) => boolean): Structs.ElementNew[] => {
+  const generator = row.section === 'colgroup' ? generators.col : generators.cell;
+  return Arr.range(amount, (idx) => Structs.elementnew(generator(), true, isLocked(idx)));
 };
 
-const rowFill = (grid: Structs.RowCells[], amount: number, generators: SimpleGenerators): Structs.RowCells[] =>
+const rowFill = (grid: Structs.RowCells[], amount: number, generators: SimpleGenerators, lockedColumns: Record<number, boolean>): Structs.RowCells[] =>
```

Comment:
This should be `lockedColumns: Record<string, boolean>` by the looks as it's passing in the result from `mapToObject`.